### PR TITLE
Make neoforge.mods.toml a generated resource, making it compatible with running from IDE

### PR DESCRIPTION
**Pull request by shartte**
Switching to this approach adds full support for placeholders in neoforge.mods.toml

In contrast to other approaches, this exposes the file where the placeholders are already replaced to the IDE as a normal source-file, meaning it is still responsible for copying it, but that also means cleaning & rebuilding from IDE has the normal intended effect, rather than it conflicting with other Gradle tasks writing to its output directory.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -156,10 +154,13 @@ tasks.withType(ProcessResources).configureEach {
...
-
-    filesMatching(['META-INF/neoforge.mods.toml']) {
-        expand replaceProperties
-    }
+    expand replaceProperties
+    from("src/main/templates")
+    into("build/generated/sources/modMetadata")
+}
+sourceSets.main.resources.srcDir(generateModMetadata)
+tasks.named("neoForgeIdeSync").configure {
```
Comment (Technici4n) [inline build.gradle:162]: Do we want to have a nice constant for this? For example, `neoForge.ideSyncTask`? Not sure we want to make the exact task name API?

File: build.gradle [hunk 2]
```diff
@@ -156,10 +154,13 @@ tasks.withType(ProcessResources).configureEach {
...
+    dependsOn generateModMetadata
```
Comment (Technici4n) [inline build.gradle:163]: Should there be brief comments after each of these 3 lines to explain what they do? Or would that be information overload?

File: build.gradle [hunk 3]
```diff
@@ -156,10 +154,13 @@ tasks.withType(ProcessResources).configureEach {
             mod_description        : mod_description
     ]
     inputs.properties replaceProperties
-
-    filesMatching(['META-INF/neoforge.mods.toml']) {
-        expand replaceProperties
-    }
+    expand replaceProperties
+    from("src/main/templates")
+    into("build/generated/sources/modMetadata")
```
Comment (Technici4n) [inline build.gradle:159]:
```suggestion
    from "src/main/templates"
    into layout.buildDirectory.dir("generated/sources/modMetadata")
```

**Review by Technici4n — COMMENTED** [review]
I'm a big fan of this change. I just converted MI to this pattern so I can finally build it Gradle-free in my IDE. :smile:

File: build.gradle (hunk 1 above)
Comment (Technici4n) [reply build.gradle:162]: I suppose that directly writing the task name in the script and exposing a constant for the task name in the API would be more idiomatic than my suggestion.

File: build.gradle (hunk 3 above)
Comment (Matyrobbrt) [reply build.gradle:157]: @shartte doesn't look that resolved to me :P

File: build.gradle [hunk 4]
```diff
@@ -156,11 +152,12 @@ tasks.withType(ProcessResources).configureEach {
...
 }
+sourceSets.main.resources.srcDir(generateModMetadata)
```
Comment (Technici4n) [inline build.gradle:159]:
```suggestion
sourceSets.main.resources.srcDir generateModMetadata
```

File: build.gradle (hunk 3 above)
Comment (Technici4n) [reply build.gradle:157]: Do it anyway. :P

Comment (Technici4n) [reply build.gradle:157]: fine, then remove the parentheses at least

**Review by Technici4n — APPROVED** [review]

**Pull request merged**